pkg/factory/cluster.go

```diff
@@ -0,0 +1,154 @@
+package factory
+
+import (
+	"github.com/giantswarm/conditions/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/conditions-handler/pkg/conditions/composite"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/controlplaneready"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/creating"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/infrastructureready"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/nodepoolsready"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/summary"
+	"github.com/giantswarm/conditions-handler/pkg/conditions/upgrading"
+	"github.com/giantswarm/conditions-handler/pkg/handler"
+)
+
+// NewClusterConditionsHandler creates a composite handler for reconciling
+// MachinePool conditions, which consists of condition handlers for
+// InfrastructureReady, ControlPlaneReady, NodePoolsReady, Ready, Creating and
+// Upgrading conditions.
+func NewClusterConditionsHandler(config handler.Config) (*composite.Handler, error) {
```

Comment:
Here we compose the conditions handler for the Cluster controller, by combining multiple smaller condition handlers, for all conditions that we can/want to set on the Cluster CR.